phac-nml/irida-next
# Attachments metadata

## What does this PR do and why?
This PR adds a metadata format tag to Attachment database entries based on the attachment file's extension.
The private function assign_metadata is executed as a ```before_create``` class method, checking by regex, if the file type is a fasta or fastq. The Attachment instance will then have ```metadata: {format: "fasta"/"fastq"}``` assigned. If the file type is neither fasta or fastq, format of ```unknown``` is assigned.

## Screenshots or screen recordings
_Screenshots are required for UI changes, and strongly recommended for all other pull requests._

## How to set up and validate locally
1. Create a project and associated sample
2. Attach a file to the sample
3. Check database entry for the Attachment's metadata field to have the corresponding format tag for the attached file type.
4. Repeat for other file types

## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.
